patches/0012-add-support-for-logging-used-Windows-APIs.patch

```diff
@@ -0,0 +1,372 @@
+ 		throw("usage")
+ 	}
++	syscallTrace(lib, &name[0])
+ 	f := stdcall2(_GetProcAddress, lib, uintptr(unsafe.Pointer(&name[0])))
+ 	return stdFunction(unsafe.Pointer(f))
+ }
+diff --git a/src/runtime/syscall_windows.go b/src/runtime/syscall_windows.go
+index 0b085835634284..2edeeb673cdec9 100644
+--- a/src/runtime/syscall_windows.go
++++ b/src/runtime/syscall_windows.go
+@@ -435,6 +435,7 @@ func syscall_loadlibrary(filename *uint16) (handle, err uintptr) {
+ 
+ //go:linkname syscall_getprocaddress syscall.getprocaddress
+ func syscall_getprocaddress(handle uintptr, procname *byte) (outhandle, err uintptr) {
++	syscallTrace(handle, procname)
+ 	outhandle, _, err = syscall_SyscallN(uintptr(unsafe.Pointer(_GetProcAddress)), handle, uintptr(unsafe.Pointer(procname)))
+ 	KeepAlive(procname)
+ 	if outhandle != 0 {
+diff --git a/src/runtime/syscalltrace_generate_windows.go b/src/runtime/syscalltrace_generate_windows.go
+new file mode 100644
+index 00000000000000..e3bbe376f43ce9
+--- /dev/null
++++ b/src/runtime/syscalltrace_generate_windows.go
+@@ -0,0 +1,46 @@
++//go:build ignore
++
++package main
++
++import (
++	"bufio"
++	"bytes"
++	"flag"
++	"log"
++	"os"
++	"strings"
++)
++
++func main() {
++	output := flag.String("output", "", "output file name")
++	flag.Parse()
++	if *output == "" {
++		log.Fatal("output file name is required")
++	}
++	out, err := os.Create(*output)
++	if err != nil {
++		log.Fatal(err)
++	}
++	defer func() {
++		if err := out.Close(); err != nil {
++			log.Fatal(err)
++		}
++	}()
++	out.WriteString("// Code generated by syscalltrace_generate_windows.go. DO NOT EDIT.\n\n")
++	out.WriteString("package runtime\n\n")
++	out.WriteString("var staticSyscalls = [...]string{\n")
++	for _, file := range flag.Args() {
++		content, err := os.ReadFile(file + ".go")
++		if err != nil {
++			log.Fatal(err)
++		}
++		scanner := bufio.NewScanner(bytes.NewReader(content))
++		for scanner.Scan() {
++			if after, found := strings.CutPrefix(scanner.Text(), "//go:cgo_import_dynamic runtime._"); found {
++				name, _, _ := strings.Cut(after, " ")
++				out.WriteString("\t\"" + name + `\x00"` + ",\n")
++			}
++		}
++	}
++	out.WriteString("}\n")
++}
+diff --git a/src/runtime/syscalltrace_windows.go b/src/runtime/syscalltrace_windows.go
+new file mode 100644
+index 00000000000000..d27e1ffeb9bffe
+--- /dev/null
++++ b/src/runtime/syscalltrace_windows.go
+@@ -0,0 +1,124 @@
++//go:generate go run syscalltrace_generate_windows.go -output zsyscalltrace_windows.go os_windows syscalltrace_windows
++
++package runtime
++
++import "unsafe"
++
++//go:cgo_import_dynamic runtime._CreateFileW CreateFileW%7 "kernel32.dll"
++//go:cgo_import_dynamic runtime._GetEnvironmentVariableW GetEnvironmentVariableW%3 "kernel32.dll"
++//go:cgo_import_dynamic runtime._GetModuleFileNameA GetModuleFileNameA%3 "kernel32.dll"
++
++var (
++	_CreateFileW,
++	_GetEnvironmentVariableW,
++	_GetModuleFileNameA,
++	_ stdFunction
++)
++
++var (
++	syscallTraceHandler uintptr
```

Comment:
This seems like a weird name for a file handle, but `syscallTraceHandle` would make sense. Typo? Or `syscallTraceFileHandle` could be even clearer.

Reply:
Type, and I like your proposal. Thanks!